main.py

```diff
@@ -394,97 +440,217 @@ def export_assets(request):
+        logging.error(f"Invalid GCS path format: {gcs_path}")
+        return "Error processing the request. Invalid GCS path format.", 400
+
+    bucket_name = parts[0]
+    object_path = parts[1]
+    resource_prefix = object_path.rsplit("/", 1)[0] + "/"
+
+    # Use GCS client to read the operation name from the file
     storage_client = storage.Client()
-    bucket = storage_client.get_bucket(data["bucket"])
+    bucket = storage_client.bucket(bucket_name)
+    blob = bucket.blob(object_path)  # Use only the object path, not the full GCS path
+    operation_name = (
+        blob.download_as_text().strip()
+    )  # ensure no leading/trailing whitespace
+    logging.info(f"Extracted operation name: {operation_name}")
+
+    # Authenticate using the default service account and create a service client for the Cloud Asset API
+    asset_service = discovery.build("cloudasset", "v1", cache_discovery=False)
+
+    # Create the request to get operation details
+    logging.info(f"Fetching details for operation: {operation_name}")
+    get_operation_request = asset_service.operations().get(name=operation_name)
+    response = get_operation_request.execute()
+
+    # Check if operation is done
+    if response.get("done", False):
+        logging.info("Asset export operation is complete. Starting file processing.")
+        # Process files in the GCS directory
+        return process_gcs_directory(bucket_name, resource_prefix)
+    else:
+        logging.warning(
+            "The asset export operation is still in progress. It will be retried."
+        )
+        raise Exception(
+            "Asset export operation not yet completed. Task will be retried."
+        )
 
-    blob = bucket.get_blob(data["name"])
 
-    # Check if blob is None
-    if blob is None:
-        logging.warning("Blob is None, returning without further action")
-        return
+def process_gcs_directory(bucket_name, prefix):
+    logging.info(
+        f"Starting to process the gcs directory with arguments: {bucket_name}, {prefix}"
+    )
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+
+    # Check if operation_name.txt exists
+    lock_blob_name = f"{prefix}operation_name.txt"
+    lock_blob = bucket.blob(lock_blob_name)
+    if not lock_blob.exists():
+        logging.info(
+            f"operation_name.txt not found at {lock_blob_name}. Exiting early."
+        )
+        return (
+            "Lockfile isn't present so assuming all files were previously processed successfully.",
+            200,
+        )
 
-    # Skip processing if filename starts with "temp"
-    if "/temp_" in blob.name:
-        logging.info("Blob name contains '/temp_', skipping processing")
-        return
+    # List blobs (i.e., files) within the GCS directory (i.e., prefix)
+    blobs = bucket.list_blobs(prefix=prefix)
+    for blob in blobs:
+        # Skip the operation_name.txt file while processing
+        if blob.name == lock_blob_name:
```

Comment:
Refactoring some of this into two smaller functions would make the code cleaner. 
Also easier to unit test the independent bits. For ex:

```
json_objects = parseBlob(blob)
if not json_objects:
  continue

asset_type, content_type = extractBlobTypes(blob)
if not asset_type:
  continue
```